File: src/sharp/Module.TerritoryData.Library/Interfaces/ITerritoryDataRepository.cs
using Module.TerritoryData.Library.Models;
using Module.TerritoryData.Library.Models.Hierarchy;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Module.TerritoryData.Library.Interfaces
{
    public interface ITerritoryDataRepository
    {
        Country GetCountry(string countryCode);
        CountryList GetCountryList();
        Level1Division GetLevel1Division(string zone1Code);
        Level1DivisionList GetLevel1DivisionList(SearchParams searchParams);
        Level2Division GetLevel2Division(string zone2Code);
        Level2DivisionList GetLevel2DivisionList(SearchParams searchParams);
        Level3Division GetLevel3Division(string zone3Code);
        Level3DivisionList GetLevel3DivisionList(SearchParams searchParams);
        City GetCity(string cityCode);
        CityList GetCityList(SearchParams searchParams);
        Address GetAddress(string addressCode);
        AddressList GetAddressList(SearchParams searchParams);

    }
}

File: src/java/module.territorydata/src/main/java/pl/digihome/module/territorydata/library/interfaces/ITerritoryDataRepository.java
package pl.digihome.module.territorydata.library.interfaces;

import pl.digihome.module.territorydata.library.models.*;
import pl.digihome.module.territorydata.library.*;

public interface ITerritoryDataRepository
{
	Country GetCountry(String countryCode);
	CountryList GetCountryList();
	Zone1 GetZone1(String zone1Code);
	Zone1List GetZone1List(SearchParams searchParams);
	Zone2 GetZone2(String zone2Code);
	Zone2List GetZone2List(SearchParams searchParams);
	Zone3 GetZone3(String zone3Code);
	Zone3List GetZone3List(SearchParams searchParams);
	City GetCity(String cityCode);
	CityList GetCityList(SearchParams searchParams);
	Address Address(String addressCode);
	AddressList GetAddressList(SearchParams searchParams);

}